Can't compile with GCC 15.2.1 20260209

I'm on Arch Linux 6.12.69-1-lts, R version 4.5.2.  Later fails to compile with the latest release of GCC (15.2.1 20260209) installed:

```
> install.packages("~/Builds/later_1.4.6.tar.gz", repos = NULL, type = "source")
Installing package into ‘/home/me/R/x86_64-pc-linux-gnu-library/4.5’
(as ‘lib’ is unspecified)
* installing *source* package ‘later’ ...
** this is package ‘later’ version ‘1.4.6’
** package ‘later’ successfully unpacked and MD5 sums checked
** using staged installation
Running configure script
-latomic linker flag not needed.
** libs
using C compiler: ‘gcc (GCC) 15.2.1 20260209’
using C++ compiler: ‘g++ (GCC) 15.2.1 20260209’
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c RcppExports.cpp -o RcppExports.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c callback_registry.cpp -o callback_registry.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c debug.cpp -o debug.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c fd.cpp -o fd.o
gcc -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects  -c init.c -o init.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c later.cpp -o later.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c later_posix.cpp -o later_posix.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c later_win32.cpp -o later_win32.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c timer_posix.cpp -o timer_posix.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c timestamp_unix.cpp -o timestamp_unix.o
g++ -std=gnu++17 -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects   -c timestamp_win32.cpp -o timestamp_win32.o
gcc -I"/usr/include/R/" -DNDEBUG -pthread -DSTRICT_R_HEADERS -DRCPP_NO_MODULES  -I'/home/me/R/x86_64-pc-linux-gnu-library/4.5/Rcpp/include' -I/usr/local/include    -fpic  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/r/src=/usr/src/debug/r -flto=auto -ffat-lto-objects  -c tinycthread.c -o tinycthread.o
In file included from tinycthread.c:25:
tinycthread.h:471:21: error: conflicting types for ‘pthread_once_t’; have ‘__once_flag’
  471 |   #define once_flag pthread_once_t
      |                     ^~~~~~~~~~~~~~
In file included from /usr/include/pthread.h:26,
                 from tinycthread.h:98:
/usr/include/bits/pthreadtypes.h:53:30: note: previous declaration of ‘pthread_once_t’ with type ‘pthread_once_t’ {aka ‘int’}
   53 | typedef int __ONCE_ALIGNMENT pthread_once_t;
      |                              ^~~~~~~~~~~~~~
make: *** [/usr/lib64/R/etc/Makeconf:202: tinycthread.o] Error 1
ERROR: compilation failed for package ‘later’
* removing ‘/home/me/R/x86_64-pc-linux-gnu-library/4.5/later’
Warning message:
In utils::install.packages("~/Builds/later_1.4.6.tar.gz", repos = NULL,  :
  installation of package ‘/home/me/Builds/later_1.4.6.tar.gz’ had non-zero exit status
```
It also fails when installing from from CRAN (and it failed when CRAN had 1.4.5 too).

I downgraded to GCC 15.2.1 20260103, and then it compiled just fine. Probably not a huge impact for most people, but wanted to point it out!

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,7 @@
 # later (development version)
 
+* Fixed #256: compilation failure with glibc >= 2.43 and GCC >= 15, caused by the C11 `once_flag` type now being defined in `<stdlib.h>` under C23. Renamed internal tinycthread symbols to avoid the namespace collision (#257).
+
 # later 1.4.6
 
 * Improved responsiveness when idle at the R console on POSIX systems (#251).

--- a/src/tinycthread.c
+++ b/src/tinycthread.c
@@ -894,7 +894,7 @@ int _tthread_timespec_get(struct timespec *ts, int base)
 #endif /* _TTHREAD_EMULATE_TIMESPEC_GET_ */
 
 #if defined(_TTHREAD_WIN32_)
-void tct_call_once(once_flag *flag, void (*func)(void))
+void tct_call_once(tct_once_flag *flag, void (*func)(void))
 {
   /* The idea here is that we use a spin lock (via the
      InterlockedCompareExchange function) to restrict access to the

--- a/src/tinycthread.h
+++ b/src/tinycthread.h
@@ -465,11 +465,11 @@ int tct_tss_set(tct_tss_t key, void *val);
   typedef struct {
     LONG volatile status;
     CRITICAL_SECTION lock;
-  } once_flag;
-  #define ONCE_FLAG_INIT {0,}
+  } tct_once_flag;
+  #define TCT_ONCE_FLAG_INIT {0,}
 #else
-  #define once_flag pthread_once_t
-  #define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
+  typedef pthread_once_t tct_once_flag;
+  #define TCT_ONCE_FLAG_INIT PTHREAD_ONCE_INIT
 #endif
 
 /** Invoke a callback exactly once
@@ -478,7 +478,7 @@ int tct_tss_set(tct_tss_t key, void *val);
  * @param func Callback to invoke.
  */
 #if defined(_TTHREAD_WIN32_)
-  void tct_call_once(once_flag *flag, void (*func)(void));
+  void tct_call_once(tct_once_flag *flag, void (*func)(void));
 #else
   #define tct_call_once(flag,func) pthread_once(flag,func)
 #endif

